decentraland/unity-explorer
# chore: streamline initialization flow - teleport - realm

## What does this PR change?

Streamlines the teleport and realm change flow. Adapts the initialization flow to use this streamline.

- Removes teleport and landscape resposibilities from `RealUserInitializationFlowController` and moves them to `RealmNavigator`. Makes the loading process as simetrical as possible, making correct usage of loading reports. Nonetheless, they dont use the same method, since `RealUserInitializationFlowController` has extra tasks not needed by `RealmNavigator` (like `RealFlowLoadingStatus`, player loading and audio handling)
- Simplifies `AsyncLoadProcessReport`. Removes the async extension
- Unifies loading controller under `ILoadingScreen` interface
- Streamlines the realm, landscape and teleport logic in `RealmNavigator`. Further changes can be done in that class for elements that need to toggle on and off when leaving worlds.
- Now you can correctly teleport between realms. Landscape handling is done correctly, it can be further extended for reusability (not part of this PR)
- You can chose a realm on startup on editor
- When going into a realm, you will now spawn on the first fixed scene pointer of the realm; and not 0,0
- Comments out `CreateEmptyPointersInFixedRealmSystem` since we now have a landscape surrounding the world

Fixes #922, #892

Possibily fixes 919, but not referencing until we can confirm it.


![image](https://github.com/decentraland/unity-explorer/assets/1999557/1a87e7e6-fb1e-4f82-a8ec-d63288ce9e79)



## How to test the changes?



1. Launch the explorer
2. Go to goerli `/world goerli`. Check that the scene is loaded before loading screen is out
3. Go to genesis `/world genesis`. Check that the scene is loaded before loading screen is out
4. Teleport to several locations and look for irregularities
5. Go to Olavra `/world olavra.dcl.eth`. Check that the scene is loaded before loading screen is out
6. From Olavra, just teleport somewhere; '/goto 100,100' for example. Check that everything keeps working as expected

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
